docs/source/deployment/amazon_emr_serverless.md

```diff
@@ -0,0 +1,242 @@
+
+**2. Create an EMR Studio**.
+
+- Go to the AWS Console > EMR > EMR Serverless.
+- Make sure you are in the correct region where you want to create your resource.
+- Then, click "Get started" > "Create and launch EMR Studio".
+
+3. **Create an application**.
+
+- Type: Spark
+- Release version: `emr-6.10.0`
+- Architecture: `x86_64`
+- Application setup options > "Choose custom settings"
+- Custom image settings > "Use the custom image with this application"
+- For simplicity, use the same custom image for both Spark drivers and executors
+- Provide the ECR image URI; it must be located in the same region as the EMR Studio
+
+You may customise other settings as required, or even change the release version and architecture options
+(along with making the changes in the custom image).
+
+```{warning}
+This code base has only been tested with the above options.
+```
+
+See the AWS EMR Serverless documentation for [more details on creating an application](https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/studio.html).
+
+### IAM
+
+1. **Create a job runtime role**. Follow the instructions under "Create a job runtime role" [here](https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/getting-started.html). This role will be used when submitting a job to EMR Serverless. The permissions defined in the policy also grant access
+to an S3 bucket (specify the S3 bucket you have created).
+
+2. **Grant access to ECR repository**.
+
+- Go to ECR in the AWS console.
+- Click into the repository you created earlier, then on the left under Repositories > Permissions.
+- Click "Edit policy JSON" and paste [this policy](https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/application-custom-image.html) under "Allow EMR Serverless to access the custom image repository".
+- Make sure to enter the ARN for your EMR Serverless application in the `aws:SourceArn` value.
+
+## (Optional) Validate the custom image
+
+EMR Serverless provides a utility to validate your custom images locally, to make sure your modifications are still
+compatible to EMR Serverless and to prevent job failures. Refer to [Amazon EMR Serverless Image CLI](https://github.com/awslabs/amazon-emr-serverless-image-cli)
+for more details.
+
+## Run a job
+
+```{note}
+On making changes to the custom image, and rebuilding and pushing to ECR, be sure to restart the EMR Serverless
+application before submitting a job if your application is **already started**. Otherwise, new changes may not be reflected in the job run.
+
+This may be due to the fact that when the application has started, EMR Serverless keeps a pool of warm resources (also referred to as
+[pre-initialized capacity](https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/application-capacity.html#pre-init-capacity))
+ready to run a job, and the nodes may have already used the previous version of the ECR image.
+```
```

Comment:
Question: this will render different on the Kedro docs just like [this one](https://docs.kedro.org/en/stable/tutorial/package_a_project.html#run-a-packaged-project), right? Just to make sure the formatting will be applied.